src/consensus/vote.cpp

```diff
@@ -545,29 +591,105 @@ bool VoteManager::pbftBlockHasEnoughValidCertVotes(SyncBlock& pbft_block_and_vot
   return valid_votes.size() >= pbft_2t_plus_1;
 }
 
-std::string VoteManager::getJsonStr(std::vector<Vote> const& votes) {
+std::string VoteManager::getJsonStr(std::vector<std::shared_ptr<Vote>> const& votes) {
   Json::Value ptroot;
   Json::Value ptvotes(Json::arrayValue);
 
-  for (Vote const& v : votes) {
+  for (auto const& v : votes) {
     Json::Value ptvote;
-    ptvote["vote_hash"] = v.getHash().toString();
-    ptvote["accounthash"] = v.getVoter().toString();
-    ptvote["sortition_proof"] = v.getSortitionProof().toString();
-    ptvote["vote_signature"] = v.getVoteSignature().toString();
-    ptvote["blockhash"] = v.getBlockHash().toString();
-    ptvote["type"] = v.getType();
-    ptvote["round"] = Json::Value::UInt64(v.getRound());
-    ptvote["step"] = Json::Value::UInt64(v.getStep());
+    ptvote["vote_hash"] = v->getHash().toString();
+    ptvote["accounthash"] = v->getVoter().toString();
+    ptvote["sortition_proof"] = v->getSortitionProof().toString();
+    ptvote["vote_signature"] = v->getVoteSignature().toString();
+    ptvote["blockhash"] = v->getBlockHash().toString();
+    ptvote["type"] = v->getType();
+    ptvote["round"] = Json::Value::UInt64(v->getRound());
+    ptvote["step"] = Json::Value::UInt64(v->getStep());
     ptvotes.append(ptvote);
   }
   ptroot["votes"] = ptvotes;
 
   return ptroot.toStyledString();
 }
 
+std::vector<std::shared_ptr<Vote>> VoteManager::getProposalVotes(uint64_t pbft_round) {
+  std::vector<std::shared_ptr<Vote>> proposal_votes;
```

Comment:
same here, reserve size based on `verified_votes_[pbft_round][1]` size

Reply:
Fixed, thanks

Reply:
move it under to lock

Reply:
yeah, fixed